controllers/bootstrap/init.go

```diff
@@ -166,19 +177,19 @@ func (b *Bootstrap) InitResources(manualManagement bool) error {
 			return err
 		}
 		// Create Operator RBAC
-		if err := b.createOrUpdateFromYaml([]byte(util.Namespacelize(operator.RBAC, b.MasterNamespace))); err != nil {
+		if err := b.createOrUpdateFromYaml([]byte(util.Namespacelize(operator.RBAC, controlNs))); err != nil {
 			return err
 		}
 		// Create Operator Deployment
-		if err := b.createOrUpdateFromYaml([]byte(util.ReplaceImages(util.Namespacelize(operator.Deployment, b.MasterNamespace)))); err != nil {
+		if err := b.createOrUpdateFromYaml([]byte(util.ReplaceImages(util.Namespacelize(operator.Deployment, controlNs)))); err != nil {
 			return err
 		}
 		// Wait for CRD ready
 		if err := b.waitResourceReady(operator.APIVersion, operator.Kind); err != nil {
 			return err
 		}
 		// Create Operator CR
-		if err := b.createOrUpdateFromYaml([]byte(util.Namespacelize(operator.CR, b.MasterNamespace))); err != nil {
+		if err := b.createOrUpdateFromYaml([]byte(util.Namespacelize(operator.CR, controlNs))); err != nil {
```

Comment:
Just a reminder, we don't need to deploy secretshare CR in the SaaS situation.
Thus I believe we can create separate functions for SaaS and non-SaaS instead of sharing the same flow. It keeps flexible when we need to remove the secretshare operator.
 
We can update it in the future PR.